pkg/crdupdater/configmap.go

```diff
@@ -29,21 +29,149 @@ import (
+	if err != nil {
+		return nil, errors.Annotatef(err, "unable to get deployer object for %s", config.namespace)
+	}
+	return &ConfigMap{
+		config:        config,
+		deployer:      deployer,
+		configMaps:    configMaps,
+		oldConfigMaps: make(map[string]*corev1.ConfigMap, 0),
+		newConfigMaps: make(map[string]*corev1.ConfigMap, 0),
+	}, nil
+}
+
+// buildNewAndOldObject builds the old and new config map
+func (c *ConfigMap) buildNewAndOldObject() error {
+	// build old config map
+	oldConfigMaps, err := c.list()
+	if err != nil {
+		return errors.Annotatef(err, "unable to get config maps for %s", c.config.namespace)
+	}
+	for _, oldConfigMap := range oldConfigMaps.(*corev1.ConfigMapList).Items {
+		c.oldConfigMaps[oldConfigMap.GetName()] = &oldConfigMap
+	}
+
+	// build new config map
+	for _, newConfigMap := range c.configMaps {
+		newConfigMapKube, err := newConfigMap.ToKube()
+		if err != nil {
+			return errors.Annotatef(err, "unable to convert config map %s to kube %s", newConfigMap.GetName(), c.config.namespace)
+		}
+		c.newConfigMaps[newConfigMap.GetName()] = newConfigMapKube.(*corev1.ConfigMap)
+	}
+
+	return nil
+}
+
+// add adds the config map
+func (c *ConfigMap) add(isPatched bool) (bool, error) {
+	isAdded := false
+	isUpdated := false
+	var err error
+	for _, configMap := range c.configMaps {
+		if _, ok := c.oldConfigMaps[configMap.GetName()]; !ok {
+			c.deployer.Deployer.AddConfigMap(configMap)
+			isAdded = true
+		} else {
+			isUpdated, err = c.patch(configMap, isPatched)
+			if err != nil {
+				return false, errors.Annotatef(err, "patch config map:")
+			}
+		}
+	}
+	if isAdded && !c.config.dryRun {
+		err = c.deployer.Deployer.Run()
+		if err != nil {
+			return false, errors.Annotatef(err, "unable to deploy config map in %s", c.config.namespace)
+		}
+	}
+	return isAdded || isUpdated, nil
+}
+
+// get gets the config map
+func (c *ConfigMap) get(name string) (interface{}, error) {
+	return util.GetConfigMap(c.config.kubeClient, c.config.namespace, name)
+}
+
+// list lists all the config maps
+func (c *ConfigMap) list() (interface{}, error) {
+	return util.ListConfigMaps(c.config.kubeClient, c.config.namespace, c.config.labelSelector)
+}
+
+// delete deletes the config map
+func (c *ConfigMap) delete(name string) error {
+	return util.DeleteConfigMap(c.config.kubeClient, c.config.namespace, name)
+}
+
+// remove removes the config map
+func (c *ConfigMap) remove() error {
+	// compare the old and new config map and delete if needed
+	for _, oldConfigMap := range c.oldConfigMaps {
+		if _, ok := c.newConfigMaps[oldConfigMap.GetName()]; !ok {
+			err := c.delete(oldConfigMap.GetName())
+			if err != nil {
+				return errors.Annotatef(err, "unable to delete config map %s in namespace %s", oldConfigMap.GetName(), c.config.namespace)
+			}
+		}
+	}
+	return nil
+}
+
+// patch patches the config map
+func (c *ConfigMap) patch(cm interface{}, isPatched bool) (bool, error) {
```

Comment:
It will be used only in replication controller and deployment type where if there is any configmap or secret updated, it will automatically patch the rc and deployment.